suprb/solution/mixing_model.py

```diff
@@ -1,28 +1,98 @@
 import numpy as np
 
 from suprb.rule import Rule
+from suprb.utils import check_random_state, RandomState
 from . import MixingModel
 
 
+class FilterSubpopulation():
+    def __init__(self, rule_amount: int = 6, random_state: RandomState = 42):
+        self.rule_amount = rule_amount
+        self.random_state = check_random_state(random_state)
+
+    def __call__(self, subpopulation: list[Rule]):
+        return subpopulation
+
+
+class NBestFitness(FilterSubpopulation):
+    def __call__(self, subpopulation: list[Rule]):
+        fitnesses = np.array([rule.fitness_ for rule in subpopulation])
+        ind = sorted(range(len(fitnesses)),
+                     key=lambda i: fitnesses[i])[-self.rule_amount:]
```

Comment:
![image](https://github.com/heidmic/suprb/assets/45732695/0e05a199-5859-48d1-b414-e9319e7b277b)

No, since we are starting to count from behind in `[-self.rule_amount:]` and from the front in `[ : self.rule_amount]`